ks3-standard/src/commonMain/kotlin/io/ks3/standard/TypeOmittingTransformer.kt

```diff
@@ -0,0 +1,30 @@
+package io.ks3.standard
+
+import kotlinx.serialization.json.JsonElement
+import kotlinx.serialization.json.JsonObject
+import kotlinx.serialization.json.JsonTransformingSerializer
+import kotlinx.serialization.json.buildJsonObject
+import kotlinx.serialization.json.jsonObject
+import kotlinx.serialization.serializer
+
+/**
+ * A serializer that omits the type discriminator when serializing with polymorphic serialization.
+ *
+ * This is useful when you want to serialize a sealed class hierarchy to JSON without including the type discriminator.
+ *
+ * @param classDiscriminatorProperty The name of the property that is used as the type discriminator. Defaults to "type".
+ */
+inline fun <reified T : Any> typeOmittingTransformer(classDiscriminatorProperty: String = "type") =
```

Comment:
Because the serial name can be altered with specific annotations, it _might_ be worth fetching the name programatically. I did that in kxs-ts-gen:

https://github.com/adamko-dev/kotlinx-serialization-typescript-generator/blob/d68dddb7cfcbe40ea96c8ffc5cfd754238c03045/modules/kxs-ts-gen-core/src/commonMain/kotlin/dev/adamko/kxstsgen/core/TsElementConverter.kt#L87-L90

So here it would look something like this:

```kotlin
  val descriptor = serializer<T>().descriptor
  
  // discriminator name
  val discriminatorIndex = descriptor.elementDescriptors
    .indexOfFirst { it.kind == PrimitiveKind.STRING }
  val discriminatorName = descriptor.elementNames.elementAtOrNull(discriminatorIndex)
```

But it's more complicated and perhaps buggy - I'll leave it up to you.

Reply:
I tried adding dynamic resolution of the discriminator. It seems that KxS adds the discriminator after transforms when you use an explicit discriminator name, so it can't be omitted by the transformer. I'll remove the option to configure the transformer and add a warning about this..